packages/apollo-server-defer/src/__tests__/starWarsDefer-test.ts

```diff
@@ -0,0 +1,552 @@
+import { isDeferredExecutionResult } from '../execute';
+import { toArray } from 'rxjs/operators';
+
+import { StarWarsSchema } from './starWarsSchema';
+import { GraphQLError } from 'graphql';
+import { graphql } from './graphql';
+
+describe('@defer Directive tests', () => {
+  describe('Basic Queries', () => {
+    it('Can @defer on scalar types', async done => {
+      const query = `
+        query HeroNameQuery {
+          hero {
+            id
+            name @defer
+          }
+        }
+      `;
+      try {
+        const result = await graphql(StarWarsSchema, query);
+        expect(isDeferredExecutionResult(result)).toBe(true);
+        if (isDeferredExecutionResult(result)) {
+          expect(result.initialResult).toEqual({
+            data: {
+              hero: {
+                id: '2001',
+              },
+            },
+          });
+          const patchesObserver = result.deferredPatchesObservable
+            .pipe(toArray())
+            .subscribe(patches => {
+              expect(patches.length).toBe(1);
+              expect(patches).toContainEqual({
+                path: ['hero', 'name'],
+                data: 'R2-D2',
+              });
+              done();
+            });
+        }
+      } catch (error) {
+        done(error);
+      }
+    });
+
+    it('Can @defer on a field on a list type', async done => {
+      const query = `
+        query HeroNameAndFriendsQuery {
+          hero {
+            id
+            name 
+            friends {
+              name @defer
+            }
+          }
+        }
+      `;
+      try {
+        const result = await graphql(StarWarsSchema, query);
+        expect(isDeferredExecutionResult(result)).toBe(true);
+        if (isDeferredExecutionResult(result)) {
+          expect(result.initialResult).toEqual({
+            data: {
+              hero: {
+                id: '2001',
+                name: 'R2-D2',
+                friends: [{}, {}, {}],
```

Comment:
this doesn't seem like a result that we want. I would have expected it to be `{ name: null }, { name: null }, { name: null }`

Reply:
Fixed